File: Colorful Stones.py
c = input()
s = input()
j, r, x = 1, 0, 2
if s[0] != c[0]:
    j, x = 0, 1
for i in s:
    if i == c[j]:
        r += 1
        j += 1
print(min(r+x, len(c)))

File: Colorful Stones.cpp
#include<bits/stdc++.h>

using namespace std;

int main()
{
    //freopen("input.txt","r",stdin);
    string c,s;
    cin >> c >> s;
    int j=1,r=0,x=2;
    if (s[0]!=c[0])
    {
        j = 0;
        x = 1;
    }

    for (int i = 0; i < s.size();++i)
    {
        if (s[i] == c[j])
        {
            r+=1;
            j+=1;
        }
    }
    int ll= c.size();
    cout << min(r+x,ll)<<"\n";
    return 0;
}
